### ISSUE TITLE
[FEATURE] Add temporal split controls for gap, test size, and max train size [M1]

### ISSUE THREAD

**Issue by dgenio** [dgenio/skdr-eval#29]
# ✨ Feature Request

## 📋 Summary
Expose richer temporal-validation controls through the public evaluation APIs, including gap, test-size, and max-train-size options.

## 🎯 Motivation
Time-aware evaluation is a core strength of the library, but real-world temporal leakage patterns often require more control than a default split strategy provides.

### Use Case
A user with highly autocorrelated or operationally clustered service logs wants an embargo gap between training and evaluation windows.

### Current Limitations
- Time-aware evaluation exists, but temporal split customization appears limited.
- Users cannot easily tune gap or train/test window shape for leakage-sensitive workloads.
- More realistic temporal validation regimes are harder to express.

## 💡 Proposed Solution
Expose `gap`, `test_size`, and `max_train_size` in standard and pairwise evaluation entry points.

### API Design
```python
import skdr_eval

report, detailed = skdr_eval.evaluate_sklearn_models(
    logs=logs,
    models=models,
    n_splits=3,
    gap=48,
    test_size=500,
    max_train_size=5000,
)
```

### Implementation Ideas
- Thread temporal controls through the split construction path.
- Validate parameter interactions and surface clear errors.
- Document when each control is appropriate.

## ✅ Acceptance Criteria
- Public evaluation APIs accept temporal split controls.
- Split behavior is covered by tests.
- Docs explain recommended use for leakage-sensitive data.
- Backward-compatible defaults remain unchanged.

## Tasks / Checklist
- [ ] Expose new temporal split parameters in public APIs
- [ ] Validate parameter combinations
- [ ] Add fold-construction tests
- [ ] Document examples and guidance

## 🔄 Alternatives Considered
- Keep a single opinionated split strategy only.
- Ask users to fork or reimplement splitting outside the library.

## 📚 Additional Context
Likely touchpoints:
- `src/skdr_eval/core.py`
- `src/skdr_eval/pairwise.py`
- `tests/test_design_and_propensity.py`
- `README.md`

### Related Papers
- TimeSeriesSplit documentation: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html
- Time-aware evaluation is central to this repo’s product promise.

### Existing Implementations
- scikit-learn TimeSeriesSplit: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html

### Benefits
- [x] Improves accuracy of estimations
- [x] Adds new evaluation capabilities

### Complexity
- [x] Medium complexity (requires some refactoring)

### Breaking Changes
- [x] No breaking changes

### Test Cases
- Gap-aware fold construction
- Fixed test-size behavior
- Max-train-size truncation

### Examples Needed
- Temporal leakage-control example in docs or quickstart extension

## 📝 Documentation Requirements
- [x] API documentation updates
- [x] Usage examples
- [x] README updates

### Priority
- [x] Medium (nice to have soon)

### Estimated Effort
- [x] Medium (few days)

## 🤝 Contribution
- [ ] I'm willing to implement this feature
- [x] I need help from maintainers

## 📎 Related Issues
- Related to support-health warnings and benchmark-harness work
